zebra-state/src/service/finalized_state.rs

```diff
@@ -390,6 +396,13 @@ impl FinalizedState {
                 batch.zs_insert(history_tree_cf, height, history_tree);
             }
 
```

Comment:
Can we calculate the new chain value balance here, using `ValueBalance::update_with_block`?

You'll need a list of UTXOs *spent* by the block.
(The previous code used UTXOs *created* by the block, which is why it didn't work.)

Here's where the UTXOs spent by this block are deleted.
You can look them up using `self.utxo(outpoint)`.

https://github.com/ZcashFoundation/zebra/blob/32e1237df85b5e5cb5617a3a0704ce26a2783f0e/zebra-state/src/service/finalized_state.rs#L332-L337